test/ibmq/test_account_client.py

```diff
@@ -248,6 +251,30 @@ def test_job_cancel(self):
                     # just retry.
                     self.assertIn('JOB_NOT_CANCELLED', str(ex))
 
+    def test_access_token_not_in_exception_traceback(self):
+        """Check that access token is replaced within chained request exceptions."""
+        backend = self.provider.backends.ibmq_qasm_simulator
+        circuit = transpile(self.qc1, backend, seed_transpiler=self.seed)
+        qobj = assemble(circuit, backend, shots=1)
+        api = backend._api
+
+        session_ = api.client_api.session
+        access_token_ = session_.access_token
+
+        exception_message = 'The access token in this exception ' \
+                            'message should be replaced: {}'.format(access_token_)
+        try:
+            with mock.patch.object(
+                    HTTPConnectionPool,
+                    'urlopen',
+                    side_effect=MaxRetryError(
+                        HTTPConnectionPool('host'), 'url', reason=exception_message)):
+                _ = api.job_submit(backend.name(), qobj.to_dict(), use_object_storage=True)
+        except RequestsApiError:
+            exception_chain_as_str = traceback.format_exc()
+            self.assertTrue(exception_chain_as_str)
+            self.assertNotIn(self.access_token, exception_chain_as_str)
```

Comment:
Can you move lines 275-276 out of the `except`? So if the assert fails, it doesn't print the previous exception (i.e. `RequestApiError`) that may contain the api token. Also instead of using `assertNotIn`, can you just check the token is not in the str and do a `self.fail()`? The `assertNotIn` would also print the access token if it fails.